.circleci/config.yml

```diff
@@ -0,0 +1,128 @@
+# iOS CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/ios-migrating-from-1-2/ for more details
+#
+version: 2
+jobs:
+  test:
+    macos:
+      xcode: "9.4.0"
+    steps: 
+      - checkout
+      - run:
+          name: tagdescription
+          command: |
+              echo "test step1"
+              echo "test step2"
+
+  release_carthage:
+    # Specify the Xcode version to use
+    macos:
+      xcode: "9.2.0"
+
+    steps: 
+      - checkout
+      - run:
+          name: upgrade carthage
+          command: brew outdated carthage || brew upgrade carthage
+      - run:
+          name: install github-release
+          command: brew install github-release
+      - run:
+          name: Build Cathage
+          command: carthage build --no-skip-current
+
+      - run:
+          name: Create Cathage Archive
+          command: carthage archive  
+
+      - run:
+          name: Rename Cathage Achive
+          command:  mv  AWSAPIGateway.framework.zip   aws-sdk-ios-carthage.framework.zip
+
+      - run:
+          name: release the tag
+          command: |
+            tagdescription=$(sed -n '/## '${CIRCLE_TAG}'/,/## [0-9]*\.[0-9]*\.[0-9]/p'  CHANGELOG.md | sed '1d' | sed '$d')
+            tagname="AWS SDK for iOS "${CIRCLE_TAG}
+            github-release release -s ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME}  -t ${CIRCLE_TAG} -d "$tagdescription" -n "$tagname"         
+      - run:
+          name: upload file to git release 
+          command: github-release upload -s ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME}  -t ${CIRCLE_TAG} -f aws-sdk-ios-carthage.framework.zip -n aws-sdk-ios-carthage.framework.zip
+
+  release_cocoapods:
+    # Specify the Xcode version to use
+    macos:
+      xcode: "9.4.0"
```

Comment:
We have to do the release-appdocs job on xcode 9.2,  as some commands used by generating documents are removed since xcode 9.3
We have to do the release-cocoapods on xocde 9.4 and later version, on which python3 is installed. 

Can we use xcode 10.0 for release cocoapods and release carthage and use xcode 9.2 for document publish

Reply:
Xcode 9.2 is an interesting callout. We should plan on phasing out appledoc if it is not going to be supported in future Xcode versions. We can do Xcode 10 and 9.2 for now 👍

Reply:
all jobs will use xcode 10.0.0